extension/asapauth/README.md

```diff
@@ -0,0 +1,30 @@
+# ASAP Client Authentication Extension
+
+This extension provides [Atlassian Service Authentication Protocol](https://s2sauth.bitbucket.io/) (ASAP) client 
+credentials for HTTP or RPC based exporters. 
+
+## Example Configuration
+
+```yaml
+extensions:
+  asapclient:
+    key_id: somekeyid
+    issuer: someissuer
+    audience:
+      - someservice
+      - someotherservice
+    private_key: ${ASAP_PRIVATE_KEY}
+    ttl_seconds: 120 # Default: 60
```

Comment:
What does this property do? Will a new token be obtained after the TTL? Would an existing token just be validated after the TTL? If this is explained in another authoritative document, like Asap's documentation, link there.

Reply:
The `exp` (expiration) claim on the JWT is set as `time.Now().Add(TTL)`. So, the token will be cached until it's expired, in which case another token will be provisioned. 

The client caching itself is implementation specific, but is related to what the spec says about token expiry, so I have linked that information here. Additionally, I have added comments to all of these yaml fields with a brief on what they are.